pipes-and-filters/PipesAndFilters.Shared/ServiceBusPipeFilter.cs

```diff
@@ -98,19 +103,17 @@ public async Task Close(TimeSpan timespan)
             //  We simply stop any new processing, wait for existing thread to complete, then close the message pump and then return
             Thread.Sleep(timespan);
 
-            this.inQueue.Close();
+            await this.processor.CloseAsync();
```

Comment:
Once, we cache the client, we we should dispose it here as well (after disposing the processor).